### ISSUE TITLE
write a basic unit test for the command line app

### ISSUE THREAD

**Issue by warengonzaga** [warengonzaga/magic-commit#10]
Summary:

There's a basic `test.js` file in the root directory; however, it is not the actual test for the current version of the command-line app. We need a tailored unit test for the command-line app.

Goal:

Write a unit test specifically for the current version and feature of the command line app.

Let me know if you want to take this issue or have questions.

**Comment by gitdevjin** [comment]
Hi, Waren. I would like to work on this, may I try this?

**Comment by Abhinav-Bansal751** [comment]
hi sir,i would also like to try this.

**Comment by warengonzaga** [comment]
Hey @gitdevjin and @Abhinav-Bansal751 feel free to submit your PR.

**Comment by gitdevjin** [comment]
@warengonzaga Thanks! :)

**Comment by warengonzaga** [comment]
Let me know which one of you can take it @Abhinav-Bansal751 @gitdevjin? So I can assign this issue.

**Comment by gitdevjin** [comment]
@warengonzaga I'll take it

**Comment by warengonzaga** [comment]
> @warengonzaga I'll take it

Done! ✨

**Comment by gitdevjin** [comment]
Hi Waren, I tried to install your tool using `npm install -g magicc` but I couldn't even run it. Maybe my Macbook's problem. I tried all the solutions that are available Online, but none of them worked for me. I also tried to install it locally following [Contributing.md](https://github.com/gitdevjin/magic-commit/blob/main/contributing.md). But it didn't work either.
I am going to try to make it work. However, if I can't figure it out until tomorrow, I'll let you know; then probably you should assign others. Thanks!

**Comment by warengonzaga** [comment]
@gitdevjin Hey there. To install it locally, you must follow the following for your tests.
1. `yarn install`
2. `yarn setup`

Since you are using a MacBook, it should work as expected because it uses Linux commands.

I wouldn't recommend installing it using npm or yarn globally.

**Comment by warengonzaga** [comment]
Otherwise, you can use CodeSpaces and do the changes there.

**Comment by Abhinav-Bansal751** [comment]
for those who are trying it in windows,and getting `rm` not recognized as internal or external command,try putting `rmdir /s /q dist` in `clean` script.This is for windows.now run  npm run build,then npm run setup.Then u can use  magicc command ig.I have Had got some peer dependency issues while doing `npm install` , and i resolved it with `npm install --legacy-peer-deps`.Hope it helps

**Comment by warengonzaga** [comment]
@Abhinav-Bansal751 might be a good addition to script in package.json file?

Then add `yarn setup:windows` command.

**Comment by Abhinav-Bansal751** [comment]
Yes ,I have added `"setup:windows":"rmdir /s /q dist && yarn build"`  in my latest commit to add context to git message PR. pls see to it.

**Comment by warengonzaga** [comment]
I think you replaced the code there; kindly take a look at my suggested changes.
https://github.com/warengonzaga/magic-commit/pull/14#discussion_r1796464222

@Abhinav-Bansal751

**Comment by gitdevjin** [comment]
@warengonzaga Yarn works, but for some reason, npm didn't work for me.
Thank you for the tips. I'll try to work on the test code :)

**Comment by gitdevjin** [comment]
@warengonzaga, thank you for the tips.
With `yarn`, I made it work. However, I ran into another problem.
First, I deleted `prettier --check .` as I think using prettier on your `README.md` and other files may not be what you want.
<img width="363" alt="Screenshot 2024-10-11 at 7 42 08 AM" src="https://github.com/user-attachments/assets/86d285b2-cc61-4e26-a533-3213b319ef4c">
Then, I ran the test `yarn test`; I got this error.
<img width="1027" alt="Screenshot 2024-10-11 at 7 43 36 AM" src="https://github.com/user-attachments/assets/3d34aa45-4470-463d-9932-bd162009a304">
I appreciate the opportunity to work on this project, but I’m finding it challenging to tackle this issue and the settings right now. I believe it might be better suited for someone else with more experience. Thank you

**Comment by warengonzaga** [comment]
> for those who are trying it in windows,and getting `rm` not recognized as internal or external command,try putting `rmdir /s /q dist` in `clean` script.This is for windows.now run npm run build,then npm run setup.Then u can use magicc command ig.I have Had got some peer dependency issues while doing `npm install` , and i resolved it with `npm install --legacy-peer-deps`.Hope it helps

I fixed this already @Abhinav-Bansal751 the easiest way to fix this is to remove those dependencies and add it again so that we can get the latest version. 😊

Because of that, it breaks our automated way to publish the NPM package.

**Comment by warengonzaga** [comment]
@gitdevjin how's your setup? Are you able to make any progress here? Let us know, I'm always here to help.

**Comment by Abhinav-Bansal751** [comment]
> > for those who are trying it in windows,and getting `rm` not recognized as internal or external command,try putting `rmdir /s /q dist` in `clean` script.This is for windows.now run npm run build,then npm run setup.Then u can use magicc command ig.I have Had got some peer dependency issues while doing `npm install` , and i resolved it with `npm install --legacy-peer-deps`.Hope it helps
> 
> I fixed this already @Abhinav-Bansal751 the easiest way to fix this is to remove those dependencies and add it again so that we can get the latest version. 😊
> 
> Because of that, it breaks our automated way to publish the NPM package.

ohk,i didnt know about that earlier.

**Comment by gitdevjin** [comment]
Hi @warengonzaga.
The application is now running; however, I'm encountering issues with the test suite. It seems that ava isn't recognizing the React syntax.

**Comment by warengonzaga** [comment]
We can remove Ava or XO there, as long as we can test if the CLI works properly.

**Comment by warengonzaga** [comment]
I'm not familiar with Ava and XO. To be honest, it came with Ink when I generated the project. According to Copilot, Ava is being used for running tests while Xo is used for lining the reaction. You can choose any other linting or test suite for our purpose. @gitdevjin

**Comment by warengonzaga** [comment]
@gitdevjin what's the progress here?

**Comment by warengonzaga** [comment]
What about using this @gitdevjin?
https://github.com/vadimdemedes/ink?tab=readme-ov-file#testing

**Comment by gitdevjin** [comment]
I apologize for the delayed response; I was occupied with midterms. I appreciate your suggestion and will give it a try. If I encounter any issues, I’ll reach out for help. Thank you for your understanding!

**Comment by warengonzaga** [comment]
No worries, @gitdevjin, just checking if you're stuck. I'm here to help.

**Comment by gitdevjin** [comment]
@warengonzaga I tried working on the test configuration issue, but I couldn’t get it to work. I think it might be best to assign it to someone else. Sorry for the trouble.

**Comment by warengonzaga** [comment]
Alright, thanks, @gitdevjin. Might someone be interested in working on this?
